### DISCUSSION TITLE
Glossary

### DISCUSSION THREAD

**Discussion by PhillypHenning** [bitovi/bitops#232] [General]
Post up any word or phrase you feel needs clarification.

**Comment by PhillypHenning** [comment]
https://github.com/bitovi/bitops/blob/plugins-doc/docs/glossary.md

**Comment by mickmcgrath13** [comment]
@PhillypHenning I'd switch the table to be "Name" and "Abbreviation(s) and Aliases" rather than "Name" and "Full name" (reason being that it'd be good to be consistent about using the full name in our writing, so maybe put that first)

Here are some terms that would be good to define (not exhaustive):

BitOps stuff
- BitOps
- Operations Repository (Ops Repo, Operations Repo)
- Operations Repo Environment (deployment environments)
- BitOps Lifecycle Hooks (hooks, before/after scripts)
- `bitops.config.yaml` (bitops config)
- `bitops.schema.yaml` (bitops schema, plugin schema)
- Tool (IaC Tools): wrapped by plugins to provide functionality
- BitOps Plugin
- Omnibus
- BitOps Base (BitOps Core)
- Base image

Software Development Terms:
- Continuous Integration (CI): build, test, and publish
- Deployment
- Continuous Deployment
- GitOps
- Pipeline
- Pipeline Runner

General terms
- Secrets
- Docker
- Docker Image
- Docker Container
- Infrastructure as Code (IaC)
- CloudFormation
- Terraform
- Ansible
- Helm
- Helm Wrapper Chart (Umbrella Chart)
- Cloud Provider
- AWS (Amazon Web Services)
- GCP (Google Cloud Platform)
- Azure (Microsoft Azure)
- Yaml (`.yaml` not `.yml`)
- CLI (command line interface)

**Reply by PhillypHenning** [reply]
Started to update the glossary with what you captured here. 

Thank you for taking the time for this

**Comment by mickmcgrath13** [comment]
issue created to carry this forward: https://github.com/bitovi/bitops/issues/258
